pkg/config/analysis/analyzers/util/hosts.go

```diff
@@ -78,7 +78,20 @@ func ConvertHostToFQDN(namespace resource.Namespace, host string) string {
 	// Convert to FQDN only if host is not a wildcard or a FQDN
 	if !strings.HasPrefix(host, "*") &&
 		!strings.Contains(host, ".") {
-		fqdn = host + "." + string(namespace) + "." + DefaultClusterLocalDomain
+		fqdn = host + "." + string(namespace) + "." + GetConfigClusterLocalDomain()
 	}
 	return fqdn
 }
+
+var configClusterLocalDomain string
+
+func SetConfigClusterLocalDomain(domain string) {
+	configClusterLocalDomain = "svc." + domain
```

Comment:
Using globals like this is not safe or acceptable, this will cause data races. note this package is used in long running servers, not just the CLI

Reply:
ok, that's indeed a problem, I'll find a new methods.